tests/test_frame.py

```diff
@@ -1398,21 +1398,125 @@ def foo(df: pd.DataFrame) -> pd.DataFrame:
             .pipe(foo)
         )
 
+    df = pd.DataFrame({"a": [1], "b": [2]})
     check(assert_type(val, pd.DataFrame), pd.DataFrame)
 
-    check(assert_type(pd.DataFrame({"a": [1]}).pipe(foo), pd.DataFrame), pd.DataFrame)
+    check(assert_type(df.pipe(foo), pd.DataFrame), pd.DataFrame)
 
     def bar(val: Styler) -> Styler:
         return val
 
-    check(
-        assert_type(pd.DataFrame({"a": [1], "b": [1]}).style.pipe(bar), Styler), Styler
-    )
+    check(assert_type(df.style.pipe(bar), Styler), Styler)
 
     def baz(val: Styler) -> str:
         return val.to_latex()
 
-    check(assert_type(pd.DataFrame({"a": [1], "b": [1]}).style.pipe(baz), str), str)
+    check(assert_type(df.style.pipe(baz), str), str)
+
+    def qux(
+        df: pd.DataFrame,
+        positional_only: int,
+        /,
+        argument_1: list[float],
+        argument_2: str,
+        *,
+        keyword_only: tuple[int, int],
+    ) -> pd.DataFrame:
+        return pd.DataFrame(df)
+
+    check(
+        assert_type(
+            # FIXME: Remove ignore when pyright releases fix:
+            # https://github.com/microsoft/pyright/pull/6797
+            df.pipe(qux, 1, [1.0, 2.0], argument_2="hi", keyword_only=(1, 2)),  # pyright: ignore[reportGeneralTypeIssues]
```

Comment:
pyright 1.1.343 is released.  Can you update `pyproject.toml` to have `pyright = ">=1.1.343` and then you can remove this?